src/server/modules/bulk/services/BulkService.ts

```diff
@@ -0,0 +1,86 @@
+import { injectable } from 'inversify'
+import paparse from 'papaparse'
+import * as interfaces from '../interfaces/BulkService'
+
+import {
+  bulkUploadMaxNum,
+  bulkUploadRandomStrLength,
+  ogHostname,
+} from '../../../config'
+import { BULK_UPLOAD_HEADER } from '../../../../shared/constants'
+import { BulkUrlMapping } from '../../../repositories/types'
+import * as validators from '../../../../shared/util/validation'
+import generateShortUrl from '../../../util/url'
+
+const BULK_UPLOAD_RANDOM_STRING_LENGTH = bulkUploadRandomStrLength
+const BULK_UPLOAD_MAX_NUM = bulkUploadMaxNum
+
+@injectable()
+export class BulkService implements interfaces.BulkService {
+  parseCsv: (dataString: string) => interfaces.CSVSchema = (dataString) => {
+    const schema = {
+      rows: 0,
+      isValid: true,
+      longUrls: [],
+    } as interfaces.CSVSchema
+
+    paparse.parse(dataString, {
+      skipEmptyLines: false,
+      delimiter: ',',
+      step(step, parser) {
+        const rowData = step.data as string[]
+        let validRow = true
+        if (schema.rows === 0) {
+          // if header is invalid
+          if (rowData[0] !== BULK_UPLOAD_HEADER) {
+            schema.isValid = false
+            parser.abort()
+          }
+        } else {
+          const acceptableLinkCount = schema.rows <= BULK_UPLOAD_MAX_NUM + 1 // rows include header
```

Comment:
real nit:
```suggestion
          const acceptableLinkCount = schema.rows < BULK_UPLOAD_MAX_NUM // rows include header
```

Reply:
Ahaha good catch I fixed this in d15791065b94f45d7fb8. To clarify the real logic should be 
`const acceptableLinkCount = schema.rows <= BULK_UPLOAD_MAX_NUM`
Takes into account the first row header